modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -21,31 +21,37 @@ def get_claims(params = {})
       }
     end
 
-    def get_claims_by_date_range(params = {})
-      validate_date_params(params['start_date'], params['end_date'])
+    def get_claims_by_date_range(params = {}) # rubocop:disable Metrics/MethodLength
+      DateTime.parse(params['start_date'].to_s) && DateTime.parse(params['end_date'].to_s)
 
       @auth_manager.authorize => { veis_token:, btsss_token: }
       faraday_response = client.get_claims_by_date(veis_token, btsss_token, params)
 
-      if faraday_response.body['data']
+      if faraday_response.status == 200
         raw_claims = faraday_response.body['data'].deep_dup
 
-        {
-          metadata: {
-            'status' => faraday_response.body['statusCode'],
-            'success' => faraday_response.body['success'],
-            'message' => faraday_response.body['message']
-          },
-          data: raw_claims&.map do |sc|
-            sc['claimStatus'] = sc['claimStatus'].underscore.titleize
-            sc
-          end
-        }
+        data = raw_claims&.map do |sc|
+          sc['claimStatus'] = sc['claimStatus'].underscore.titleize
+          sc
+        end
+
+        Faraday::Response.new(
+          response_body: { 'statusCode' => faraday_response.body['statusCode'], 'success' => true,
+                           'message' => faraday_response.body['message'], 'data' => data }, 'status' => 200
+        )
+
       end
       # Because we're appending this to the Appointments object, we need to not just throw an exception
-      # TODO: Integrate error handling from the token client through every subsequent client/service
-    rescue Faraday::Error
-      nil
+    rescue Date::Error => e
+      Rails.logger.debug(message: "#{e}. (given: #{params['start_date']} & #{params['end_date']}).")
+      Faraday::Response.new(response_body: {
+                              'statusCode' => 400,
+                              'message' => "#{e}. (given: #{params['start_date']} & #{params['end_date']}).",
+                              'success' => false
+                            }, status: 400)
+    rescue => e
```

Comment:
I don't know that we should rescue any exception here. We already rescue any Faraday::Errors above, as well as Date::Errors. I want any other errors to be as noisy as possible. What do you think?

Reply:
This is why it stopped working - after I tested it and confirmed it would work, I reworked it to only rescue `Faraday::Error`s, that's what broke it. 🫤 

The issue is that this method is _only_ for the appt > claim association service (because of the way we're handling errors), so we can't just throw an exception, we have to wrap it and forward it to the association service so we can append it to the appt list, regardless of what kind of error/exception it is.

Reply:
> The issue is that this method is only for the appt > claim association service

I wonder then if we should move this method _to_ the `ClaimAssociationService`